### DISCUSSION TITLE
Where should I save the pem file in the serverless structure?

### DISCUSSION THREAD

**Discussion by Chuncheonian** [alexandercerutti/passkit-generator#91] [Q&A] [answered]
Hi~
I have a question while building a serverless.

Where should I keep the pem file?
S3?

I'm not good at English, but I hope you understand.
Thanks!

https://github.com/alexandercerutti/passkit-generator/blob/3326fc29d7ee19092df0500adb2665543bf85da1/examples/serverless/src/shared.ts#L74

**Comment by alexandercerutti** [comment, marked as answer]
Hey there @Chuncheonian, thanks for using passkit-generator!
Yes, your question is clear. I'm not an expert of serverless and similar, but AWS and other systems should have a "safe", where you can store your keys and access them through `process.env`.

For example, I think, https://aws.amazon.com/kms/ or https://docs.aws.amazon.com/lambda/latest/dg/configuration-envvars.html

Hope this is useful.

**Comment by Chuncheonian** [comment]
@alexandercerutti 

Oh, I know that I could change the pem file to string.

Thanks to you, I built a server and a serverless successfully.

Thanks for your attention, `passkit-generator`
